lib/IDE/Refactoring.cpp

```diff
@@ -5734,15 +5744,23 @@ class AsyncConverter : private SourceEntityWalker {
   void addHandlerCall(const CallExpr *CE) {
     auto Exprs = TopHandler.extractResultArgs(CE);
 
+    bool addedReturnOrThrow = true;
     if (!Exprs.isError()) {
-      OS << tok::kw_return;
+      // It's possible the user has already written an explicit return statement
+      // for the completion handler call, e.g 'return completion(args...)'. In
+      // that case, be sure not to add another return.
+      auto *parent = getWalker().Parent.getAsStmt();
```

Comment:
Getting sillier and sillier, but... what if we had `return (completion(arg))`? In that case the parent will be a `ParenExpr`. Not sure there's a better way to handle that than a flag on entering/leaving a `ReturnStmt` though.

I'm probably okay getting this in regardless since it fixes the most common case + the crashing.

Reply:
I *believe* that we won't transform `(completion(arg))` because we check for `NestedExprCount == 0`. So we'd likely need some extra handling to dig into the semantic expr to both transform it and then avoid adding an extra return. I can add a test either way to make sure we at least capture the current behaviour though

Reply:
Ah true - yeah, add test with FIXME I think.